presto-tests/src/main/java/com/facebook/presto/tests/DistributedQueryRunner.java

```diff
@@ -336,7 +336,12 @@ public void waitForClusterToGetReady()
         int availableCoordinators = 0;
         while (availableCoordinators != coordinators.size()) {
             MILLISECONDS.sleep(10);
-            availableCoordinators = getResourceManager().get().getNodeManager().getCoordinators().size();
+            if (getResourceManager().isPresent()) {
+                availableCoordinators = getResourceManager().get().getNodeManager().getCoordinators().size();
+            }
+            if (getCatalogServer().isPresent()) {
+                availableCoordinators = getCatalogServer().get().getNodeManager().getCoordinators().size();
+            }
```

Comment:
Can you explain a little bit on this logic?

Reply:
Yes, so [here](https://github.com/prestodb/presto/blob/ac31f746650cdeffa9ed54a0a7a0f005bf1695b4/presto-tests/src/test/java/com/facebook/presto/tests/tpch/TpchQueryRunner.java#L95) waits for the node to get ready before starting. Before only the resource manager node is waited for meaning that if I start the catalog server instead it wont look for that as well. What im doing is also waiting for the catalog server if enabled so that we can run it without the resource manager

Reply:
Does it mean if both catalog and resource manager are enabled, and catalog is fully started but resource manager is not, it will not wait for resource manager anymore, which seems like a bug?

Reply:
No, in short the code before will wait for all of the coordinators to be active before letting the query runner start. Meaning that whether we start both or only one it doesn't matter as long as all of the coordinators are now active. It uses the resource manager node as a means to find the amount of active nodes. Adding this condition will make sure that if only catalog server is added it will wait as well. If both are being used then it wont change anything, the wait will still be until all coordinators are active